### PR TITLE
Merge 4.45 code freeze to develop

### PR DESCRIPTION
**Pull request by oguzkocer**

### CODE REVIEW

File: Simplenote/Resources/AppStoreStrings.pot [hunk 1]
```diff
@@ -15,12 +15,14 @@ msgstr ""
 
 #. translators: App name to be displayed in the Apple App Store. Limit to 30 characters including spaces and commas!
 msgctxt "app_store_title"
-msgid "Simplenote – Notes and Todos"
+msgid "Simplenote – Notes and Todos
```
Comment (jkmassel) [inline Simplenote/Resources/AppStoreStrings.pot]: This _may_ break the ability to import this file into GlotPress? Might be worth testing it with https://github.com/wordpress-mobile/docker-glotpress-linter?

File: Simplenote/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -1,857 +0,0 @@
-/* Number of found search results */
```
Comment (jkmassel) [inline Simplenote/en.lproj/Localizable.strings]: It seems surprising we would delete this entire file? I think something might've gone wrong here

**Review by jkmassel — CHANGES_REQUESTED** [review]

**Comment by oguzkocer** [comment]
@jkmassel Thanks for the review. Any chance you could help me with these issues? I have no idea what went wrong. The script didn't log any errors, so I thought everything was fine, but I agree that this seems off.

**Comment by mokagio** [comment]
I doubt either of you would be working on this right now because of time zones, but just dropping a message to say that I found the localization issue and I'll be submitting a fix soon.

**Comment by mokagio** [comment]
@jkmassel @oguzkocer @AliSoftware the file got deleted because we now have `./Simplenote/Supporting Files/main.m` which matches the query for files to feed as input to `genstrings` but, because of the space in the path, makes the command fail, with cascading results.

I opened two PRs with different takes on the fix:

- https://github.com/Automattic/simplenote-ios/pull/1436 updates `localize.py` to be more robust against spaces in paths
- https://github.com/Automattic/simplenote-ios/pull/1437 takes a different approach and removes the space from the path

~~I'm not sure which is best. Maybe both should be applied?~~ Actually, after further reviewing the diffs (see the "Compare" links in the description for both PRs) I was surprised to see that the output `.strings` file is different 🤔 The diff in #1437 seems more trust-worthy in that the other has a lot of deleted entries 🤔 

I'd be leaning towards using that for now and then spend more time investigating what's going on in `localize.py`. Maybe I was too hasty with the fix and I'd need to dive deeper in how the script uses `genstrings`.

**Comment by mokagio** [comment]
Also notice that the UI tests are failing and that seems to be a consistent problem since a couple of days... You can see that they are failing on [a PR based of `develop`](https://github.com/Automattic/simplenote-ios/pull/1438), too, so I think we can ignore that failure here.

I'm following up on the issue, which I think is actually backend related, elsewhere.

**Comment by AliSoftware** [comment]
@mokagio thanks for looking into it and for trying out the 2 solutions!

> Actually, after further reviewing the diffs (see the "Compare" links in the description for both PRs) I was surprised to see that the output .strings file is different 🤔 

I took a stab at it especially to try to understand why there was a difference in output in the 2 alternatives. I finally found the issue in your #1437 fix (missing parentheses to apply `-print0` to both `.m` and `.swift`), and after fixing that, #1437 ended up with the same diff for the generated `.lproj` as the diff in #1436.

Now that #1437 is fixed, I think we should go with that option (as opposed to #1436), as supporting spaces (instead of avoiding them) will be more resilient. Besides, as you said, `Supporting Files` (with a space) is indeed a common and standard name for this folder anyway, so there's a chance it could be accidentally changed back in the future by someone not realizing the consequence otherwise.

**Comment by mokagio** [comment]
Thanks @AliSoftware 🙌 

As you can guess from my merge of #1437 into this branch, I do agree with your rationale (besides, this is yet another easy to change decision, so let's move ahead!)

File: Simplenote/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -163,7 +171,7 @@
 "Collaborators" = "Collaborators";
 
 /* No comment provided by engineer. */
-"collaborators-description" = "Add an email address to share this note with someone. Then you can both make changes to it.";
+"collaborators-description" = "collaborators-description";
```
Comment (mokagio) [inline Simplenote/en.lproj/Localizable.strings]: I'm concerned by this and I think we should fix it before merging:

Screen Shot 2021-09-22 at 11 13 58 am

I feel like it's an issue that existed already and that the improvement on how we run `genestrings` from #1437 revealed. 

Looking at the source, there is no comment in the `NSLocalizedString` macro:

https://github.com/Automattic/simplenote-ios/blob/bfa530c86e3cbd4fe89c01666de9f39351e83221/Simplenote/Classes/SPAddCollaboratorsViewController.m#L148

**Review by mokagio — COMMENTED** [review]
I noticed the change in localization tooling resulted in a few strings changing in a way that is not correct.

We should address this before merging.

File: Simplenote/en.lproj/Localizable.strings (hunk 2 above)
Comment (mokagio) [reply Simplenote/en.lproj/Localizable.strings]: Fixed by #1437.

File: Simplenote/Resources/AppStoreStrings.pot (hunk 1 above)
Comment (mokagio) [reply Simplenote/Resources/AppStoreStrings.pot]: I didn't test what the linter said. I just went ahead and fixed it in [`0a9932f` (#1435)](https://github.com/Automattic/simplenote-ios/pull/1435/commits/0a9932f73d5fb5b31d0aa4d347087dcd9acd10b9)

**Comment by mokagio** [comment]
> We should address [the generated strings not being human readable] before merging.

I opened https://github.com/Automattic/simplenote-ios/pull/1443 for this.

**Review by jkmassel — APPROVED** [review]

**Pull request merged**